### PR TITLE
rework TCK to Junit5

### PR DESCRIPTION
**Pull request by aserkes**
rework TCK to Junit5, replace System.out/err.println by java.util.logging.Logger

Signed-off-by: aserkes <andrii.serkes@oracle.com>

### CODE REVIEW

**Review by lukasj — COMMENTED** [review]
LGTM, @scottmarlow what do you think?

**Comment by scottmarlow** [comment]
I noticed that this removes Arquillian which I think was added via https://github.com/eclipse-ee4j/jsonp/pull/232.  Why are removing Arquillian when I think Arquillian supports JUnit5 now?

File: tck/tck-tests/src/main/java/jakarta/jsonp/tck/api/pointertests/PointerTests.java [hunk 1]
```diff
@@ -33,14 +27,8 @@
  * JSON-P API defines {@link jakarta.json.JsonPointer} interface to work with RFC
  * 6901 JSON Pointer.
  */
-@RunWith(Arquillian.class)
```
Comment (lukasj) [inline tck/tck-tests/src/main/java/jakarta/jsonp/tck/api/pointertests/PointerTests.java:36]: this should have stayed - some tests need to keep running on the server

Comment (lukasj) [reply tck/tck-tests/src/main/java/jakarta/jsonp/tck/api/pointertests/PointerTests.java:36]: hm, my bad, this does not test server-specific functionality.

**Comment by lukasj** [comment]
@scottmarlow I believe first step was to move tests from the platform to this project `as is`, this second step is to simplify them. I believe that at least 99% tests do not require server-like environment and even that 1%, if it is there, can be rewritten to not require it.

@aserkes correct me if I'm wrong

**Review by lukasj — APPROVED** [review]
LGTM

**Comment by scottmarlow** [comment]
@lukasj 
> @scottmarlow Can't the example by mail be followed? There is standalone TCK for jakarta mail (mail-tck repo) and jakarta mail specific platform-level tests see https://github.com/eclipse-ee4j/jakartaee-tck/tree/master/src/com/sun/ts/tests/javamail

Yes, that is the `both` approach IMO, regardless of how it is accomplished (via Java Test Harness, Junit and/or something else).

@jbescos 
> I was discussing this with @lukasj and we understand this in different way. Lets clarify what is the meaning of running junit tests on `ejb jsp servlet appclient`.
> 
> This is my understanding:
> 
> Some TCKs need to deploy some war or ear files in a servlet container (glassfish, weblogic, etc). In this case, the ant script was compiling and deploying the code in the sevlet container before the TCK tests starts to run.
> 
> Then, the TCK tests make some requests to the server, then the server run any operation and returns back the result to the TCK. Then TCK evaluates it and checks whether fails or success.
> 
> Few observations from this:
> 
>     * Server is never running TCKs, this is done by the client. There is no need to have JUnit or harness dependency in the server.

Agreed that server is never running TCKs but test harness is needed in the server currently for some TCKs.  https://github.com/eclipse-ee4j/jakartaee-tck/blob/master/user_guides/jakartaee/src/main/jbake/content/troubleshooting.adoc mentions  this (as does https://github.com/eclipse-ee4j/jakartaee-tck/blob/master/user_guides/jaxrs/src/main/jbake/content/config.inc).  Look for `tsharness.jar` in either file (also mentioned in a few other user guide doc files).  

> 
>     * Some projects are lucky (like jakarta activation). They don't depend on deployments in the server. We don't need to care about servlet containers in this case, right?.

Correct, I'm only raising a concern with new component TCKs that contain tests that we were discussing removing from the Platform tck.  The concern that I meant to raise is that Jakarta EE Platform Specification requirements need to still be validated in some way that meets the expectation of Jakarta EE Platform team/committers.  

So, with the current jsonp TCK approach of not using Arquillian, I think we should leave enough jsonp tests in the Platform TCK to validate that json works as expected for the various Jakarta EE Platform vehicles.  

Is it possible for the Platform TCK jsonp tests to do its validation using the new JSONP SPEC component TCK test classes?  If yes, then in setting that up in the Platform TCK, it seems that the duplicate classes could be deleted from the Platform TCK.  

In terms of how the user running the Platform TCK will obtain the correct JSONP SPEC component TCK that contains those test classes is to be worked out.  Perhaps the Platform TCK could include a JSONP pom.xml that specifies the https://download.eclipse.org/jakartaee/jsonp/2.1/jakarta-jsonp-tck-2.1.0.zip but the user could manually update to a newer `jakarta-jsonp-tck-2.1.x` version as JSONP service releases become available.  This JSONP pom.xml in the Platform TCK could copy json TCK artifact(s) to the Platform TCK lib for the Platform TCK JSONP tests to use.

**Comment by lukasj** [comment]
@scottmarlow

> The current Platform requirements are already clearly stated in the Platform SPEC

whole chapter 6 is about requirements. _How_ the platform TCK wants to validate these requirements for specs providing standalone TCKs? Or is standalone TCK required to validate platform-level requirements?

> ...we need the JSONP SPEC TCK to be complete...

That's the plan. To achieve TCK completion, one needs to know _how_ the platform wants to validate requirements from the chapter 6 in the platform spec and/or what the platform TCK needs from the standalone TCK in order to be able to perform such validation. Does the platform require Arquillian based test(s) to be available? 

>> And more important - how? Isn't "duplicity" - no deletion from the platform TCK - way to go for EE 10?

> Possible outcomes from deletion of all Platform JSONP tests are (in no particular order):

I did not ask for removal of existing tests from the jakartaee-tck repo. I was asking if keeping them "duplicated" in the platform TCK - in the other words not removing them (or at least some of them) - for EE 10 does solve the problem with validation of the platform level requirement. If so, the question of using Arquillian in the standalone version of the TCK is, IMO, irrelevant.

**Review by lukasj — APPROVED** [review]
LGTM

**Pull request merged**